### DISCUSSION TITLE
Discussion on implementing TLS in DiceDB

### DISCUSSION THREAD

**Discussion by roopeshsn** [DiceDB/dice#1302] [General]
Hi, Team! I am going through TLS to implement in DiceDB for a couple of weeks now. As mentioned in issue #549, I found ways for implementation except at the 3rd point. I am thinking to start with AsyncServer (internal/server/server.go), as it will be invoked if `enable-multithreading` is false (single-threaded). The current implementation of AsyncServer is that it uses functions from the `syscall` package (to read and write the TCP payload) rather than the `net` package (`net.Listen` for example). `tls.Listen` uses `net.Listen` to listen to the port. I am not sure how I can introduce `tls.Listen()` to the existing listener:
https://github.com/DiceDB/dice/blob/03debc1203d0084e5527927d5a590e8707ff6fc3/internal/server/server.go#L151

Your help and guidance is highly appreciated! @JyotinderSingh @soumya-codes

**Comment by JyotinderSingh** [comment]
Thanks for picking this up @roopeshsn. Please note that we're going to deprecate async server very soon. It would be better if you design the feature wrt the resp server (the multi-threaded one).
